pkg/agent/core/state.go

```diff
@@ -922,6 +974,8 @@ func (s *State) Debug(enabled bool) {
 }
 
 func (s *State) UpdatedVM(vm api.VmInfo) {
+	hadLFCMetrics := *s.internal.scalingConfig().EnableLFCMetrics
```

Comment:
Why not just do 
```
if !*s.internal.scalingConfig().EnableLFCMetrics {
     s.internal.LFCMetrics = nil
}
```
i.e. zeroing metrics when we detect "LFC is disabled", not "it was disabled, but became enabled".

Reply:
Fixed in 5a65920df49af712b6e6a4271e87293a109ab8ad.